src/main/java/bio/terra/service/upgrade/Migrate.java

```diff
@@ -1,361 +1,59 @@
- * deployment. If not in Kubernetes, it will be based on a random UUID.
- * - take a postgres lock on the deployment table to serialize instances running this algorithm
- * - read the deployment row
- * - If the id = the KubeService id, we return false - we are not the first through with that deployment id.
- * - If the id != the KubeService id and the row is locked, we spin until the row is unlocked and return false.
- * TODO: validate that the locker is still alive by listing pods
- * - If the id != the KubeService id and the row is unlocked:
- * -- update the deployment row with the new deployment and mark that this pod owns the lock
  * -- perform the migrate according to the configuration settings
- * -- return true
  * We rely on JobService to call back in to:
  * -- update the deployment row, unlocking the row and releasing any waiting DRmanagers
  * That is because we need to hold the migration lock during Stairway migration.
  * <p>
  * This is vulnerable to failure: if this pod crashes still holding the deployment lock, we will be stuck and
  * have to clear it by hand.
  * <p>
- * If we decide that is an important case to cover, we can fix it by doing yet another check that the lock holder
- * is on the list of running pods. For now, not gonna do it.
- * <p>
- * NOTE: you might be wondering why this code does not use JdbcTemplate. I wonder why also.
- * I initially coded it with JdbcTemplate and the @Transactional annotation got the error:
- * PSQLException: ERROR: LOCK TABLE can only be used in transaction blocks
- * I tried a number of solutions and learned some Spring quirks on the way. My guess is that it has
- * something to do with the timing of the PlatformTransactionManager configuration with respect to
- * execution of the StartupInitializer. In the end, I never solved the problem.
- * I converted the code to use java.sql directly with explicit transactions.
  */
 @Component
 public class Migrate {
     private static final Logger logger = LoggerFactory.getLogger("bio.terra.service.upgrade");
     private final DataRepoJdbcConfiguration dataRepoJdbcConfiguration;
     private final MigrateConfiguration migrateConfiguration;
-    private final KubeService kubeService;
-    private final DataSource dataSource;
-
-    private enum MigrateAction {
-        NOTHING,
-        WAIT_FOR_UNLOCK,
-        MIGRATE
-    }
-
-    private enum WaitState {
-        KEEP_WAITING,
-        DONE_WAITING,
-        RETRY_MIGRATE
-    }
-
-    private static class DeploymentRow {
-        private String id;
-        private String lockingPodName;
-
-        public String getId() {
-            return id;
-        }
-
-        public DeploymentRow id(String id) {
-            this.id = id;
-            return this;
-        }
 
-        public String getLockingPodName() {
-            return lockingPodName;
-        }
-
-        public DeploymentRow lockingPodName(String lockingPodName) {
-            this.lockingPodName = lockingPodName;
```

Comment:
Is there a concern that if we don't lock, multiple pods starting could interfere with each other?

Reply:
Ah I should have mentioned in the description, I'm currently looking at the test runner pod deletion test to check that this doesn't fall over with multiple pods

Reply:
Liquibase has its own built in lock system which disallows simultaneous migrations. Is there an additional benefit we'd get by adding our own locking too?

Reply:
Not at all!  I didn't realize that they had their own (though that makes sense)

Reply:
Oh I misunderstood the question...yes the liquibase locking should prevent the pods from interfering with each other. My concern was whether always running the migration would cause start-up delays but I think since we were already locking pods right now there shouldn't be much of a difference here.